bootstrap/main.py

```diff
@@ -0,0 +1,130 @@
+
+
+def stop_old_container() -> None:
+    """ Looks for an old core container and stops it if found"""
+    try:
+        old_container = client.containers.get(CORE_CONTAINER_NAME)
+        old_container.stop()
+        old_container.remove()
+    except Exception:
+        pass
+
+
+def overwrite_config_file_with_defaults() -> None:
+    """Overwrites the config file with the default configuration"""
+    shutil.copy(default_file_path, config_file_path)
+
+
+def read_config_file() -> Any:
+    """Tries to read the config file
+
+    Returns:
+        Any: Json data of startup.json
+    """
+
+    # Tries to open the current file
+    try:
+        with open("/config/startup.json") as config_file:
+            return json.load(config_file)
+    except Exception:
+        # Copy defaults over and read again
+        shutil.copy(default_file_path, config_file_path)
+        with open("/config/startup.json") as config_file:
+            return json.load(config_file)
+
+
+def start_core() -> None:
+    """Loads core settings and launches the core docker. Loads default settings if no settings are found"""
+    core_version = "stable"
+
+    config = read_config_file()
+
+    try:
+        image = config["core"]["image"]
+        core_version = config["core"]["tag"]
+        binds = config["core"]["binds"]
+        ports = config["core"]["ports"]
+        privileged = config["core"]["privileged"]
+        network = config["core"]["network"]
+
+    except Exception as error:
+        print(f"Error reading startup json data! {error}")
+        sys.exit(1)
+
+    print("Starting core... this could take a while if the image is not available offline...")
+    client.containers.run(
+        f"{image}:{core_version}",
+        name=CORE_CONTAINER_NAME,
+        volumes=binds,
+        ports=ports,
+        privileged=privileged,
+        network=network,
+        detach=True,
+    )
+    print("Started, open http://localhost:8080")
+
+
+def core_is_running() -> bool:
+    """
+    Returns:
+        bool: True if the core container is running
+    """
+    for container in client.containers.list():
+        if CORE_CONTAINER_NAME in container.name:
+            return True
+    return False
+
+
+def remove_core() -> None:
+    """Deletes the core container if it exists (needed for updating the running image)"""
+    try:
+        old_container = client.containers.get(CORE_CONTAINER_NAME)
+        old_container.remove()
+    except Exception:
+        # This exception is raised if the container does not exist
+        pass
+
+
+if __name__ == "__main__":
+    while True:
+        if core_is_running():
+            print("core is already running, waiting for it to stop...")
+            time.sleep(1)
+        else:
+            try:
+                remove_core()
+                start_core()
+                print("Done")
+                sys.exit(0)
+            except Exception as error:
+                print(f"error: {error}, restarting...")
```

Comment:
it'll not restart, it'll try to remove the core again, no ?

Reply:
it goes back to line 119, I could reword that. "retrying"?